crates/bevy_dev_tools/src/fps_overlay.rs

```diff
@@ -0,0 +1,136 @@
+use bevy_app::{Plugin, Startup, Update};
+use bevy_asset::AssetServer;
+use bevy_color::Color;
+use bevy_diagnostic::{DiagnosticsStore, FrameTimeDiagnosticsPlugin};
+use bevy_ecs::{
+    component::Component,
+    query::With,
+    schedule::{common_conditions::resource_changed, IntoSystemConfigs},
+    system::{Commands, Query, Res, Resource},
+};
+use bevy_text::{Text, TextSection, TextStyle};
+use bevy_ui::node_bundles::TextBundle;
+
+#[derive(Default)]
+/// A plugin that adds an FPS overlay to the Bevy application.
+pub struct FpsOverlayPlugin {
```

Comment:
I think we should warn the user that this tool will effectively add the `FrameTimeDiagnosticsPlugin` if not added, maybe someone may find it surprising?